#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.14509988491951925 -0.9412954547263055 -0.30480959681076353 5.4592569131998845
  position -21.47608591724632 -8.996989322559116 23.04687784172102
  followType "None"
}
Background {
  skyColor [
    0.388235 0.662745 1
  ]
}
Solid {
  children [
    Solid {
      translation -2.5 0 0.5
      rotation 0 1 0 1.5708
      children [
        Shape {
          appearance DEF walls MattePaint {
            baseColor 0 0.156863 0.231373
          }
          geometry DEF wallPlane Plane {
            size 1 15
          }
        }
      ]
      name "Wall"
      boundingObject USE wallPlane
    }
    Solid {
      rotation 0 1 0 0
      children [
        Shape {
          appearance MattePaint {
            baseColor 0.0784314 0.0784314 0.0784314
          }
          geometry DEF groundPlane Plane {
            size 5 15
          }
        }
      ]
      name "Ground"
      boundingObject USE groundPlane
    }
    Solid {
      translation 2.5 0 0.5
      rotation 0.70710528118436 3.3905113482557537e-09 -0.707108281185553 3.14159
      children [
        Shape {
          appearance USE walls
          geometry DEF wallPlane Plane {
            size 1 15
          }
        }
      ]
      name "Wall(1)"
      boundingObject USE wallPlane
    }
    Solid {
      translation 0 7.5 0.5
      rotation 0.5773489358556708 0.5773509358554485 -0.5773509358554485 2.0944
      children [
        Shape {
          appearance USE walls
          geometry DEF wallPlane Plane {
            size 1 5
          }
        }
      ]
      name "Wall(2)"
      boundingObject USE wallPlane
    }
    Solid {
      translation 0 -7.5 0.5
      rotation 0.5773476025217157 -0.5773516025189619 -0.5773516025189619 -2.094395307179586
      children [
        Shape {
          appearance USE walls
          geometry DEF wallPlane Plane {
            size 1 5
          }
        }
      ]
      name "Wall(3)"
      boundingObject USE wallPlane
    }
  ]
  name "Stadium"
}
Solid {
  translation 0 -2.52 0.5
  children [
    Shape {
      appearance MattePaint {
        baseColor 1 0 0
      }
      geometry DEF obstacleBox Box {
        size 1 1 1
      }
    }
  ]
  name "obstacleBox"
  boundingObject USE obstacleBox
  physics Physics {
    density 0.01
  }
}
Solid {
  translation -0.29 2.99 0.5
  children [
    Shape {
      appearance MattePaint {
        baseColor 1 0 0
      }
      geometry DEF obstacleBox Box {
        size 1 1 1
      }
    }
  ]
  name "obstacleBox(1)"
  boundingObject USE obstacleBox
  physics Physics {
    density 0.01
  }
}
Solid {
  translation 0.99 0.14 0.5
  children [
    Shape {
      appearance MattePaint {
        baseColor 1 0 0
      }
      geometry DEF obstacleBox Box {
        size 1 1 1
      }
    }
  ]
  name "obstacleBox(2)"
  boundingObject USE obstacleBox
  physics Physics {
    density 0.01
  }
}
Solid {
  translation -1.76 0.19 0.5
  children [
    Shape {
      appearance MattePaint {
        baseColor 1 0 0
      }
      geometry DEF obstacleBox Box {
        size 1 1 1
      }
    }
  ]
  name "obstacleBox(3)"
  boundingObject USE obstacleBox
  physics Physics {
    density 0.01
  }
}
Robot {
  translation 0 -6.06 0.1
  children [
    DistanceSensor {
      translation 0.05 0.2 0
      rotation 0 0 1 1.309
      name "rightTouchSensor"
      lookupTable [
        0 1 0
        0.5 0 0
      ]
    }
    DistanceSensor {
      translation -0.05 0.2 0
      rotation 0 0 1 1.8326
      name "leftTouchSensor"
      lookupTable [
        0 1 0
        0.5 0 0
      ]
    }
    DistanceSensor {
      translation 0 -0.2 0
      rotation -0.577349935856137 0.577349935856137 0.5773509358560258 -2.094395307179586
      name "backTouchSensor"
      lookupTable [
        0 1 0
        0.5 0 0
      ]
    }
    LED {
      translation -0.18 0 0.05
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.03 0.01 0.03
          }
        }
      ]
      name "leftMotorIndicator"
      color []
      gradual TRUE
    }
    LED {
      translation 0.18 0 0.05
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.03 0.01 0.03
          }
        }
      ]
      name "rightMotorIndicator"
      color []
      gradual TRUE
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor -0.212 0 -0.005
      }
      device [
        RotationalMotor {
          name "leftMotor"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation -0.212 0 -0.005
        rotation 1.4327067372385162e-15 1 0 1.5707999999999944
        children [
          DEF wheel Shape {
            appearance Rubber {
            }
            geometry DEF wheelGeom Cylinder {
              height 0.02
              radius 0.05
            }
          }
        ]
        name "wheelLeft"
        boundingObject USE wheelGeom
        physics Physics {
          density 1
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0.212 0 -0.005
      }
      device [
        RotationalMotor {
          name "rightMotor"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation 0.212 0 -0.005
        rotation 1.4327067372385162e-15 1 0 1.5707999999999944
        children [
          DEF wheel Shape {
            appearance Rubber {
            }
            geometry DEF wheelGeom Cylinder {
              height 0.02
              radius 0.05
            }
          }
        ]
        name "wheelLeft(1)"
        boundingObject USE wheelGeom
        physics Physics {
          density 1
        }
      }
    }
    Shape {
      appearance MattePaint {
        baseColor 0 0.0470588 0.929412
      }
      geometry DEF puck_body Cylinder {
        height 0.1
        radius 0.2
      }
    }
  ]
  name "puck cam"
  boundingObject USE puck_body
  physics Physics {
    density 1
  }
  controller "simpleLearningController"
  supervisor TRUE
}
Robot {
  translation -1.44 -6.06 0.13
  children [
    DistanceSensor {
      translation 0.05 0.2 0
      rotation 0 0 1 1.309
      name "rightTouchSensor"
      lookupTable [
        0 1 0
        0.5 0 0
      ]
    }
    DistanceSensor {
      translation -0.05 0.2 0
      rotation 0 0 1 1.8326
      name "leftTouchSensor"
      lookupTable [
        0 1 0
        0.5 0 0
      ]
    }
    DistanceSensor {
      translation 0 -0.2 0
      rotation -0.577349935856137 0.577349935856137 0.5773509358560258 -2.094395307179586
      name "backTouchSensor"
      lookupTable [
        0 1 0
        0.5 0 0
      ]
    }
    LED {
      translation -0.18 0 0.05
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.03 0.01 0.03
          }
        }
      ]
      name "leftMotorIndicator"
      color []
      gradual TRUE
    }
    LED {
      translation 0.18 0 0.05
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.03 0.01 0.03
          }
        }
      ]
      name "rightMotorIndicator"
      color []
      gradual TRUE
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor -0.212 0 -0.005
      }
      device [
        RotationalMotor {
          name "leftMotor"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation -0.212 0 -0.005
        rotation 1.4327067372385162e-15 1 0 1.5707999999999944
        children [
          DEF wheel Shape {
            appearance Rubber {
            }
            geometry DEF wheelGeom Cylinder {
              height 0.02
              radius 0.05
            }
          }
        ]
        name "wheelLeft"
        boundingObject USE wheelGeom
        physics Physics {
          density 1
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0.212 0 -0.005
      }
      device [
        RotationalMotor {
          name "rightMotor"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation 0.212 0 -0.005
        rotation 1.4327067372385162e-15 1 0 1.5707999999999944
        children [
          DEF wheel Shape {
            appearance Rubber {
            }
            geometry DEF wheelGeom Cylinder {
              height 0.02
              radius 0.05
            }
          }
        ]
        name "wheelLeft(1)"
        boundingObject USE wheelGeom
        physics Physics {
          density 1
        }
      }
    }
    Shape {
      appearance MattePaint {
        baseColor 0 0.0470588 0.929412
      }
      geometry DEF puck_body Cylinder {
        height 0.1
        radius 0.2
      }
    }
  ]
  name "puck cam(1)"
  boundingObject USE puck_body
  physics Physics {
    density 1
  }
  controller "simpleLearningController"
  supervisor TRUE
}
Robot {
  translation 1.08 -6.06 0.16
  children [
    DistanceSensor {
      translation 0.05 0.2 0
      rotation 0 0 1 1.309
      name "rightTouchSensor"
      lookupTable [
        0 1 0
        0.5 0 0
      ]
    }
    DistanceSensor {
      translation -0.05 0.2 0
      rotation 0 0 1 1.8326
      name "leftTouchSensor"
      lookupTable [
        0 1 0
        0.5 0 0
      ]
    }
    DistanceSensor {
      translation 0 -0.2 0
      rotation -0.577349935856137 0.577349935856137 0.5773509358560258 -2.094395307179586
      name "backTouchSensor"
      lookupTable [
        0 1 0
        0.5 0 0
      ]
    }
    LED {
      translation -0.18 0 0.05
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.03 0.01 0.03
          }
        }
      ]
      name "leftMotorIndicator"
      color []
      gradual TRUE
    }
    LED {
      translation 0.18 0 0.05
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.03 0.01 0.03
          }
        }
      ]
      name "rightMotorIndicator"
      color []
      gradual TRUE
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor -0.212 0 -0.005
      }
      device [
        RotationalMotor {
          name "leftMotor"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation -0.212 0 -0.005
        rotation 1.4327067372385162e-15 1 0 1.5707999999999944
        children [
          DEF wheel Shape {
            appearance Rubber {
            }
            geometry DEF wheelGeom Cylinder {
              height 0.02
              radius 0.05
            }
          }
        ]
        name "wheelLeft"
        boundingObject USE wheelGeom
        physics Physics {
          density 1
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0.212 0 -0.005
      }
      device [
        RotationalMotor {
          name "rightMotor"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation 0.212 0 -0.005
        rotation 1.4327067372385162e-15 1 0 1.5707999999999944
        children [
          DEF wheel Shape {
            appearance Rubber {
            }
            geometry DEF wheelGeom Cylinder {
              height 0.02
              radius 0.05
            }
          }
        ]
        name "wheelLeft(1)"
        boundingObject USE wheelGeom
        physics Physics {
          density 1
        }
      }
    }
    Shape {
      appearance MattePaint {
        baseColor 0 0.0470588 0.929412
      }
      geometry DEF puck_body Cylinder {
        height 0.1
        radius 0.2
      }
    }
  ]
  name "puck cam(2)"
  boundingObject USE puck_body
  physics Physics {
    density 1
  }
  controller "simpleLearningController"
  supervisor TRUE
}
Ball {
  translation -0.67 -0.94 1.09
  radius 0.5
}
Ball {
  translation 0 0.81 1.09
  name "ball(1)"
  radius 0.5
}
Ball {
  translation 0 -4.5 1.09
  name "ball(2)"
  radius 0.5
}
Ball {
  translation 0 5.33 1.09
  name "ball(3)"
  radius 0.5
}
